File: TorneosWeb/src/app/models/enrollment-request.ts
import { Team } from './team';
import { Tournament } from './tournament';

export type RequestStatus = 'Pending' | 'Approved' | 'Rejected';

export interface NewTeamPlayer {
  firstName: string;
  lastName: string;
  jerseyNumber: number;
}

export interface EnrollmentRequest {
  id: number;
  teamId: number | null;
  tournamentId: number;
  requesterEmail: string;
  status: RequestStatus;
  createdAt: string;
  team?: Team;
  tournament?: Tournament;
  newTeamName?: string;
  newTeamCaptainName?: string;
  newTeamLogoUrl?: string;
  newTeamStadiumId?: number;
  newTeamPlayersJson?: string;
}

export function parseNewTeamPlayers(json?: string): NewTeamPlayer[] {
  if (!json) return [];
  try { return JSON.parse(json); } catch { return []; }
}

File: Torneos.API/Entities/EnrollmentRequest.cs
namespace Torneos.API.Entities;

public class EnrollmentRequest
{
    public int Id { get; set; }

    // Existing team request
    public int? TeamId { get; set; }
    public Team? Team { get; set; }

    // New team request (used when TeamId is null)
    public string? NewTeamName { get; set; }
    public string? NewTeamCaptainName { get; set; }
    public string? NewTeamLogoUrl { get; set; }
    public int? NewTeamStadiumId { get; set; }
    public string? NewTeamPlayersJson { get; set; }

    public int TournamentId { get; set; }
    public Tournament Tournament { get; set; } = null!;
    public string RequesterEmail { get; set; } = string.Empty;
    public string Status { get; set; } = "Pending";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
